setup.py

```diff
@@ -32,24 +32,18 @@
     "grpcio >= 1.38.1, < 2.0dev",  # https://github.com/googleapis/python-bigquery/issues/695
     "google-api-core[grpc] >= 1.29.0, < 3.0.0dev",
     "proto-plus >= 1.10.0",
+    "google-cloud-bigquery-storage >= 2.0.0, <3.0.0dev",
     "google-cloud-core >= 1.4.1, < 3.0dev",
     "google-resumable-media >= 0.6.0, < 3.0dev",
     "packaging >= 14.3",
     "protobuf >= 3.12.0",
+    "pyarrow >= 1.0.0, < 5.0dev",
```

Comment:
Making these previously-optional dependencies now required feels like it might be a `semver: major` change.

Reply:
IIRC @tswast said in a chat that it shouldn't be a problem for most users, and those who cannot install these dependencies for any reason can still pin to a lower (non-major version). But reading this as I write, we might actually want to re-consider this.

Reply:
There has been a [debate](https://github.com/googleapis/python-bigquery/issues/757#issuecomment-879908013) on this on the issue itself. Also copying Tim's input:

> The gist is that the problematic dependencies are already required transitively anyway
With pyarrow being the exception
It's probably worth the bump for pyarrow if we're being conservative about it. Esp since we do know of environments where adding it will break

Reply:
Let's go with the major version bump. I'd rather people adopt this change a little more slowly than break someone who has a working 2.x branch without pyarrow

Reply:
Roger.

Is there perhaps anything else that we want to clean up and would be a breaking change? (possibly in a separate PR)
A major version bump is a good opportunity for that.

Reply:
https://github.com/googleapis/python-bigquery/issues/319 is still on my mind. Perhaps easier now that proto-plus added more case options?

Reply:
Another breaking change we might want to do: https://issuetracker.google.com/144712110 -- Better type conversions in the pandas connector.

Reply:
And yet another one from today - maybe bumping the minimum `pyarrow` version, as the older ones suffer from a [bug](https://github.com/googleapis/python-bigquery/issues/781#issuecomment-883497414) (load data from dataframe can upload incorrect data, yikes!)

Reply:
That is a nasty bug. Thankfully `pyarrow` didn't follow [NEP-29](https://numpy.org/neps/nep-0029-deprecation_policy.html) this time and kept support for Python 3.6 in the 4.x packages. I'd be okay with bumping the minimum for that.

Reply:
And yet another - if this gets delayed significantly (read: months), we might want to consider dropping Python 3.6 support as well. But Python 3.6 [end of life](https://www.python.org/dev/peps/pep-0494/#id15) will only happen in December this year.